### PR TITLE
Add our own Vector2 type.

### PR DESCRIPTION
**Pull request by PJB3005**

### CODE REVIEW

File: SS14.Shared/Maths/Vector2.cs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        ///     Clamps the components of a vector to minimum and maximum vectors.
+        /// </summary>
+        /// <param name="vector">The vector to clamp.</param>
+        /// <param name="min">The lower bound vector.</param>
+        /// <param name="max">The upper bound vector.</param>
+        public static Vector2 Clamp(Vector2 vector, Vector2 min, Vector2 max)
+        {
+            return new Vector2(
+                vector.X.Clamp(min.X, max.X),
+                vector.X.Clamp(min.Y, max.Y)
```
Comment (YotaXP) [inline SS14.Shared/Maths/Vector2.cs]: Shouldn't this be `vector.Y.Clamp(...)`?

Comment (PJB3005) [reply SS14.Shared/Maths/Vector2.cs]: Yes it definitely should be. Good catch.

**Review by psykzz — APPROVED** [review]

File: SS14.Shared/Maths/Vector2.cs [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+        /// <summary>
+        ///     Returns the hash code for this instance.
+        /// </summary>
+        /// <returns>A unique hash code for this instance.</returns>
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                return (X.GetHashCode() * 397) ^ Y.GetHashCode();
```
Comment (psykzz) [inline SS14.Shared/Maths/Vector2.cs:261]: why the random number here?

Comment (Acruid) [reply SS14.Shared/Maths/Vector2.cs:261]: It is a prime number that reduces collisions in the hash.

Comment (gbasood) [reply SS14.Shared/Maths/Vector2.cs:261]: Still, it's a magic number. Make it a const somewhere, and comment on why it exists so we don't have people asking this question a year from now.

**Pull request merged**